test/run-spec-wasm2c.py

```diff
@@ -211,15 +250,19 @@ def _WriteCommand(self, command):
 
     def _WriteModuleCommand(self, command):
         self.module_idx += 1
-        self.out_file.write('%s_init();\n' % self.GetModulePrefix())
+        self.out_file.write('%s_init_module();\n' % self.GetModulePrefix())
+        self.out_file.write('%s_module_instance_t %s;\n' % (self.GetModulePrefix(), self.GetModuleInstanceName()))
+        self._WriteModuleInstanceImports(command, False)
 
     def _WriteModuleCleanUps(self):
-        for idx in range(1, self.module_idx):
-            self.out_file.write("%s_free();\n" % self.GetModulePrefix(idx))
+        for idx in range(0, self.module_idx):
```

Comment:
Why does this change from 1 to 0?

Reply:
This was my mistake in #1901; it wasn't caught in testing (by the asan memory leak detector) until the table stopped being a global data structure. (We were failing to free module zero...)

Reply:
Ah I see.  In that case maybe this can just be `range(self.module_idx)`... and maybe a separate bugfix PR if you want to land it right away?

Reply:
Sounds good, done here and will submit bugfix PR.